### PR TITLE
Working single-node distributed training of Qwen3-30B-A3B-Base with DeepSpeed

### PR DESCRIPTION
**Pull request by shaltielshmid**
I create a new directory called `qwen_model_shaltiel`, and copied in files from `qwen_model` as I needed them, and modifying them. I mostly kept the original `train.py`, and just modified it a bit. 

I put a lot of work into the README file in that directory (no LLMs involved), please review it in depth, specifically the ## Next Steps section.  @sarelWE @sarelWeinberger @dnrevital

### CODE REVIEW

**Comment by shaltielshmid** [comment]
I'm not surprised - this is the warmup stage with a very low LR, on a sample with a global batch size of 1. That's also why the loss is spikey. 
I just wanted to see results. 
If we increase the GBS (either by increasing the MBS or increasing the GAS (gradient accumulation steps), the loss should become less spikey. 
And as it progresses, say 1000-2000 steps, then we expect the loss to start decreasing. Obviously dependent on max steps and the warmup, assuming the warmup finishing by 50% through.

**Comment by dnrevital** [comment]
Sarel's plan is to train each dataset 500 steps and take the training-loss decrease as a quality measure for each dataset. It seems that we need either to increase these 500 to at least 1500-2000 to start seeing something useful, or find the way to make decrease earlier.
